source/client-side-encryption/client-side-encryption.md

```diff
@@ -0,0 +1,2483 @@
+
+If the [extraOptions.cryptSharedLibRequired](#extraoptions.cryptsharedlibrequired) option is `true` then the driver MUST
+NOT attempt to spawn or connect to `mongocryptd`.
+
+> \[!NOTE\]
+>
+> Since spawning [mongocryptd](#mongocryptd) requires checking whether [crypt_shared](#crypt_shared) is loaded, and
+> checking whether [crypt_shared](#crypt_shared) is available can only be done *after* having initialized the
+> `libmongocrypt_handle`, drivers will need to defer spawning [mongocryptd](#mongocryptd) until *after* initializing
+> [libmongocrypt](#libmongocrypt) and checking for [crypt_shared](#crypt_shared).
+
+### Spawning [mongocryptd](#mongocryptd)
+
+If a MongoClient is configured for Client Side Encryption (eg. bypassAutoEncryption=false), then by default (unless
+mongocryptdBypassSpawn=true), mongocryptd MUST be spawned by the driver. Spawning MUST include the command line argument
+--idleShutdownTimeoutSecs. If the user does not supply one through extraOptions.mongocryptdSpawnArgs (which may be
+either in the form "--idleShutdownTimeoutSecs=60" or as two consecutive arguments `["--idleShutdownTimeoutSecs", 60]`,
+then the driver MUST append --idleShutdownTimeoutSecs=60 to the arguments. This tells mongocryptd to automatically
+terminate after 60 seconds of non-use. The stdout and stderr of the spawned process MUST not be exposed in the driver
+(e.g. redirect to /dev/null). Users can pass the argument --logpath to extraOptions.mongocryptdSpawnArgs if they need to
+inspect mongocryptd logs.
+
+Upon construction, the MongoClient MUST create a MongoClient to mongocryptd configured with
+serverSelectionTimeoutMS=10000.
+
+If spawning is necessary, the driver MUST spawn mongocryptd whenever server selection on the MongoClient to mongocryptd
+fails. If the MongoClient fails to connect after spawning, the server selection error is propagated to the user.
+
+### Connecting to [mongocryptd](#mongocryptd)
+
+If the [crypt_shared](#crypt_shared) library is loaded, the driver MUST NOT attempt to connect to
+[mongocryptd](#mongocryptd). (Refer: [Detecting crypt_shared Availability](#detecting-crypt_shared-availability)).
+
+Single-threaded drivers MUST connect with
+[serverSelectionTryOnce=false](../server-selection/server-selection.md#serverselectiontryonce) , connectTimeoutMS=10000,
+and MUST bypass [cooldownMS](../server-discovery-and-monitoring/server-discovery-and-monitoring.rst#cooldownms) when
+connecting to mongocryptd. See
+[Why are serverSelectionTryOnce and cooldownMS disabled for single-threaded drivers connecting to mongocryptd?](#why-are-serverselectiontryonce-and-cooldownms-disabled-for-single-threaded-drivers-connecting-to-mongocryptd).
+
+If the ClientEncryption is configured with mongocryptdBypassSpawn=true, then the driver is not responsible for spawning
+mongocryptd. If server selection ever fails when connecting to mongocryptd, the server selection error is propagated to
+the user.
```

Comment:
```suggestion
Single-threaded drivers MUST connect with
[serverSelectionTryOnce=false](../server-selection/server-selection.md#serverselectiontryonce), `connectTimeoutMS=10000`,
and MUST bypass [cooldownMS](../server-discovery-and-monitoring/server-discovery-and-monitoring.rst#cooldownms) when
connecting to mongocryptd. See
[Why are serverSelectionTryOnce and cooldownMS disabled for single-threaded drivers connecting to mongocryptd?](#why-are-serverselectiontryonce-and-cooldownms-disabled-for-single-threaded-drivers-connecting-to-mongocryptd)

If the ClientEncryption is configured with `mongocryptdBypassSpawn=true`, then the driver is not responsible for spawning
mongocryptd. If server selection ever fails when connecting to mongocryptd, the server selection error is propagated to
the user.
```

Didn't apply backticks to the linked terms, since they already stand out.